File: abelkhan_type_component/protcol/csharp/xor_key.cs
using System;
using System.Collections;

namespace abelkhan
{
/*this enum code is codegen by abelkhan codegen for c#*/

/*this struct code is codegen by abelkhan codegen for c#*/
/*this caller code is codegen by abelkhan codegen for c#*/
/*this cb code is codegen by abelkhan for c#*/
    public class rsp_cb_xor_key : abelkhan.Imodule {
        public rsp_cb_xor_key(abelkhan.modulemng modules) : base("rsp_cb_xor_key")
        {
            modules.reg_module(this);

        }
    }

    public class xor_key_caller : abelkhan.Icaller {
        public rsp_cb_xor_key rsp_cb_xor_key_handle;
        public xor_key_caller(abelkhan.Ichannel _ch, abelkhan.modulemng modules) : base("xor_key", _ch)
        {
            rsp_cb_xor_key_handle = new rsp_cb_xor_key(modules);
        }

        public void refresh_xor_key(UInt32 xor_key){
            var _argv_f4b8900f_a3d1_11ea_aca6_a85e451255ad = new ArrayList();
            _argv_f4b8900f_a3d1_11ea_aca6_a85e451255ad.Add(xor_key);
            call_module_method("refresh_xor_key", _argv_f4b8900f_a3d1_11ea_aca6_a85e451255ad);
        }

    }
/*this module code is codegen by abelkhan codegen for c#*/
    public class xor_key_module : abelkhan.Imodule {
        private abelkhan.modulemng modules;
        public xor_key_module(abelkhan.modulemng _modules) : base("xor_key")
        {
            modules = _modules;
            modules.reg_module(this);

            reg_method("refresh_xor_key", refresh_xor_key);
        }

        public delegate void cb_refresh_xor_key_handle(UInt32 xor_key);
        public event cb_refresh_xor_key_handle onrefresh_xor_key;

        public void refresh_xor_key(ArrayList inArray){
            var _xor_key = (UInt32)inArray[0];
            if (onrefresh_xor_key != null){
                onrefresh_xor_key(_xor_key);
            }
        }

    }

}

File: abelkhan_type_component/protcol/ts/xor_key.ts
import abelkhan = require("../../abelkhan_type/ts/abelkhan");
import { v1 as uuidv1 } from 'uuid'
/*this enum code is codegen by abelkhan codegen for ts*/

/*this struct code is codegen by abelkhan codegen for typescript*/
/*this caller code is codegen by abelkhan codegen for typescript*/
/*this cb code is codegen by abelkhan for ts*/
export class rsp_cb_xor_key extends abelkhan.Imodule {
    constructor(modules:abelkhan.modulemng){
        super("rsp_cb_xor_key");
        modules.reg_module(this);

    }
}

export class xor_key_caller extends abelkhan.Icaller {
    public rsp_cb_xor_key_handle : rsp_cb_xor_key;
    constructor(_ch:any, modules:abelkhan.modulemng){
        super("xor_key", _ch);
        this.rsp_cb_xor_key_handle = new rsp_cb_xor_key(modules);
    }

    public refresh_xor_key(xor_key:number){
        let _argv_e8ba6c00_9f46_11ea_b8b7_a85e451255ad:any[] = [];
        _argv_e8ba6c00_9f46_11ea_b8b7_a85e451255ad.push(xor_key);
        this.call_module_method("refresh_xor_key", _argv_e8ba6c00_9f46_11ea_b8b7_a85e451255ad);
    }

}
/*this module code is codegen by abelkhan codegen for typescript*/
export class xor_key_module extends abelkhan.Imodule {
    private modules:abelkhan.modulemng;
    constructor(modules:abelkhan.modulemng){
        super("xor_key");
        this.modules = modules;
        this.modules.reg_module(this);

        this.reg_method("refresh_xor_key", this.refresh_xor_key.bind(this));
        this.cb_refresh_xor_key = null;

    }

    public cb_refresh_xor_key : (xor_key:number)=>void | null;
    refresh_xor_key(inArray:any[]){
        let _argv_e8bae130_9f46_11ea_8f3b_a85e451255ad:any[] = [];
        _argv_e8bae130_9f46_11ea_8f3b_a85e451255ad.push(inArray[0]);
        if (this.cb_refresh_xor_key){
            this.cb_refresh_xor_key.apply(null, _argv_e8bae130_9f46_11ea_8f3b_a85e451255ad);
        }
    }

}
